client/source/js/modules/ui/index.js

```diff
@@ -50,19 +50,47 @@ define([
                   angular
                     .element('<input type="file">')
                     .change(function (event) {
-                      uploadDataSpreadsheet(event.target.files[0]);
+                      uploadData(event.target.files[0], '/api/project/update');
                     })
                     .click();
                   } else {
-                      modalService.inform(
-                        function (){ },
-                        'Okay',
-                        'Create or open a project first.',
-                        'Cannot proceed'
-                      );
+                    showProjectNotActive();
                   }
                 }
-              }
+              },
+              {
+                title: 'Download project',
+                click: function () {
+                  if (activeProject.isSet()) {
+                    $http({url:'/api/project/downloadproject/'+ activeProject.name,
+                      method:'GET',
+                      headers: {'Content-type': 'application/json'},
+                      responseType:'arraybuffer'})
+                      .success(function (response, status, headers, config) {
+                        var blob = new Blob([response], { type: 'application/json' });
+                        saveAs(blob, (activeProject.name + '.json'));
+                      })
+                      .error(function (response) {});
+                  } else {
+                    showProjectNotActive();
+                  }
+                }
+              },
```

Comment:
The card stated that we should first alert the user that this operation will wipe all existing data. Not done.